src/Entity/EncryptedFieldValue.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * @file
+ * Contains \Drupal\field_encrypt\Entity\EncryptedFieldValue.
+ */
+
+namespace Drupal\field_encrypt\Entity;
+
+use Drupal\Core\Field\BaseFieldDefinition;
+use Drupal\Core\Entity\ContentEntityBase;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\field\Entity\FieldStorageConfig;
+
+/**
+ * Defines the EncryptedFieldValue entity.
+ *
+ * @ingroup field_encrypt
+ *
+ * @ContentEntityType(
+ *   id = "encrypted_field_value",
+ *   label = @Translation("Encrypted field value"),
+ *   base_table = "encrypted_field",
+ *   data_table = "encrypted_field_data",
+ *   render_cache = FALSE,
+ *   admin_permission = "administer encrypted_field_value entity",
+ *   fieldable = FALSE,
+ *   translatable = TRUE,
+ *   entity_keys = {
+ *     "id" = "id",
+ *     "uuid" = "uuid",
+ *     "langcode" = "langcode"
+ *   },
+ * )
+ */
+class EncryptedFieldValue extends ContentEntityBase implements EncryptedFieldValueInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function baseFieldDefinitions(EntityTypeInterface $entity_type) {
+    $fields = parent::baseFieldDefinitions($entity_type);
+
+    $fields['entity_type'] = BaseFieldDefinition::create('string')
+      ->setLabel(t('Entity type'))
+      ->setDescription(t('The entity type for which to store the encrypted value.'))
+      ->setSetting('is_ascii', TRUE)
+      ->setSetting('max_length', EntityTypeInterface::ID_MAX_LENGTH);
+
+    $fields['entity_id'] = BaseFieldDefinition::create('entity_reference')
+      ->setLabel(t('Entity ID'))
+      ->setDescription(t('The ID of the entity for which to store the encrypted value.'))
+      ->setRequired(TRUE);
+
+    $fields['entity_revision_id'] = BaseFieldDefinition::create('integer')
+      ->setLabel(t('Revision ID'))
+      ->setDescription(t('The revision ID of the entity.'))
+      ->setSetting('unsigned', TRUE);
+
+    $fields['field_name'] = BaseFieldDefinition::create('string')
+      ->setLabel(t('Field name'))
```

Comment:
It seems a requirement to me.
When you have an entity with 5 encrypted text fields, you wouldn't know which encrypted value is linked to what field on the entity, if we left out a reference to the fieldname.